src/plugin_example.rs

```diff
@@ -0,0 +1,148 @@
+extern crate alumet;
+extern crate anyhow;
+extern crate humantime_serde;
+extern crate serde;
+// ANCHOR: all
+use alumet::{
+    measurement::{MeasurementAccumulator, MeasurementPoint, Timestamp},
+    metrics::TypedMetricId,
+    pipeline::{elements::error::PollError, trigger::TriggerSpec, Source},
+    plugin::{
+        rust::{deserialize_config, serialize_config, AlumetPlugin},
+        AlumetPluginStart, AlumetPostStart, ConfigTable,
+    },
+    resources::{Resource, ResourceConsumer},
+    units::{PrefixedUnit, Unit, UnitPrefix},
+};
+use serde::{Deserialize, Serialize};
+use std::{fs::File, io::Read, time::Duration};
+
+// ANCHOR: Config
+#[derive(Serialize, Deserialize, Debug)]
+struct Config {
+    #[serde(with = "humantime_serde")]
+    poll_interval: Duration,
+}
+// ANCHOR_END: Config
+// ANCHOR: Metrics
+pub struct Metrics {
+    a_metric: TypedMetricId<u64>,
+}
+// ANCHOR_END: Metrics
+// ANCHOR: MyPlugin_Struct
+pub struct MyPlugin {
+    config: Config,
+    metrics: Option<Metrics>,
+}
+// ANCHOR_END: MyPlugin_Struct
+// ANCHOR: impl_default_config
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            poll_interval: Duration::from_secs(1),
+        }
+    }
+}
+// ANCHOR_END: impl_default_config
+// ANCHOR: MyPluginSource
+#[derive(Debug)]
+struct MyPluginSource {
+    random_byte: TypedMetricId<u64>,
+}
+// ANCHOR_END: MyPluginSource
+// ANCHOR: implAlumetPlugin
+impl AlumetPlugin for MyPlugin {
+    // So we define the name of the plugin.
+    fn name() -> &'static str {
+        "MyPlugin"
+    }
+
+    // We also define it's version.
+    fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+
+    // We use the default config by default and on initialization.
+    fn default_config() -> anyhow::Result<Option<ConfigTable>> {
+        Ok(Some(serialize_config(Config::default())?))
+    }
+
+    // We also use the default config on initialization and we deserialize the config
+    // to take in count if there is a different config than the default one.
+    fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
+        let config = deserialize_config(config)?;
+        Ok(Box::new(MyPlugin {
+            config,
+            metrics: None,
+        }))
+    }
+
+    // The start function is here to register metrics, sources and output.
+    fn start(&mut self, alumet: &mut AlumetPluginStart) -> anyhow::Result<()> {
+        // ANCHOR: createPrefixedUnit
+        let my_byte_unit: PrefixedUnit = PrefixedUnit {
+            base_unit: Unit::Byte,
+            prefix: UnitPrefix::Plain,
+        };
+        // ANCHOR_END: createPrefixedUnit
```

Comment:
Actually you don't need that. You can pass a plain unit to Alumet:
```
alumet.create_metric(..., Unit::Byte, ...)
```

Reply:
it's for educational purpose, I separate both the unit creation and the metric creation. Do you still want me to merge them ?

Reply:
Well, I believe that a tutorial should show good practices, and using `PrefixedUnit` with `UnitPrefix::Plain` is not idiomatic. The idiomatic thing to do here is to use `Unit::Byte` directly, and to let Rust can `into()` on it.